diff/parse.go

```diff
@@ -436,12 +435,6 @@ func (r *HunksReader) ReadHunk() (*Hunk, error) {
 			r.hunk.Body = append(r.hunk.Body, '\n')
 		}
 	}
-
-	// Final hunk is complete. But if we never saw a hunk in this call to ReadHunk, then it means we hit EOF.
-	if r.hunk != nil {
-		return r.hunk, nil
-	}
-	return nil, io.EOF
```

Comment:
Yes, it compiles in Go 1.4 and later (maybe 1.3 too, I forget when they improved unreachable code detection). Travis would fail if it did not compile.